plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts

```diff
@@ -0,0 +1,123 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PluginEndpointDiscovery } from '@backstage/backend-common';
+import { Entity } from '@backstage/catalog-model';
+import { IndexableDocument, DocumentCollator } from '@backstage/search-common';
+import fetch from 'cross-fetch';
+import unescape from 'lodash/unescape';
+import { Logger } from 'winston';
+
+export interface TechDocsDocument extends IndexableDocument {
+  kind: string;
+  namespace: string;
+  entityKind: string;
+  entity: string;
+  lifecycle: string;
+}
+
+export class DefaultTechDocsCollator implements DocumentCollator {
+  protected discovery: PluginEndpointDiscovery;
+  protected locationTemplate: string;
+  private readonly logger: Logger;
+  public readonly type: string = 'tech-docs';
+
+  constructor({
+    discovery,
+    locationTemplate,
+    logger,
+  }: {
+    discovery: PluginEndpointDiscovery;
+    locationTemplate?: string;
+    logger: Logger;
+  }) {
+    this.discovery = discovery;
+    this.locationTemplate =
+      locationTemplate || '/docs/:namespace/:kind/:name/:path';
+    this.logger = logger;
+  }
+
+  async execute() {
+    const catalogBaseUrl = await this.discovery.getBaseUrl('catalog');
+    const techDocsBaseUrl = await this.discovery.getBaseUrl('techdocs');
+    const res = await fetch(`${catalogBaseUrl}/entities`);
+    const entities: Entity[] = await res.json();
+    const docPromises = entities
+      .filter(it => it.metadata?.annotations?.['backstage.io/techdocs-ref'])
+      .map(
+        async (entity: Entity): Promise<TechDocsDocument[]> => {
+          const entityInfo = {
+            kind: entity.kind,
+            namespace: entity.metadata.namespace || 'default',
+            name: entity.metadata.name,
+          };
+
+          try {
+            const searchIndexResponse = await fetch(
+              DefaultTechDocsCollator.constructDocsIndexUrl(
+                techDocsBaseUrl,
+                entityInfo,
+              ),
+            );
+            const searchIndex = await searchIndexResponse.json();
+
+            return searchIndex.docs.map((doc: TechDocsDocument) => ({
+              title: unescape(doc.title),
+              text: unescape(doc.text || ''),
+              location: this.applyArgsToFormat(this.locationTemplate, {
+                namespace: entity.metadata.namespace || 'default',
+                kind: entity.kind,
+                name: entity.metadata.name,
+                path: doc.location,
+              }),
+              kind: 'Documentation',
```

Comment:
I wonder if this should better contain the kind of the entity that the techdocs page is related to (so `entityKind`)? Choosing the same field name would allow to query unified search results and provide matching UI elements on the search page.

Different types of "search results" can already be distinguished by the document type.